week-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,24 @@
-function setAlarm() {}
+
+function setAlarm() {
+  const setTimeTo = document.querySelector("#alarmSet");
+  const countdown = document.querySelector("#timeRemaining");
+  let time = Number(setTimeTo.value);
+
+  function titleTime(){
+    const minutes = Math.floor(time/60);
+    const seconds = time % 60;
+    countdown.innerText = `Time remaining: ${minutes.toString().padStart(2,"0")}:${seconds.toString().padStart(2,"0")}`
+
+    if (time===0){
+      playAlarm();
+      clearInterval(intervalId);
+    }
+
+    time = time -1;    
+  }
+  titleTime();
+  const intervalId = setInterval(titleTime,1000);
+}
```

Comment:
Would you be able to try to refactor it a bit? Try to declare `titleTime` function outside of `setAlarm` function? It is a good practice to declare functions separately if possible.